src/osd/ECBackend.cc

```diff
@@ -1024,11 +1026,35 @@ shard_to_read_map.find(from);
 
   assert(rop.in_progress.count(from));
   rop.in_progress.erase(from);
+  bool is_complete = true;
   if (!rop.in_progress.empty()) {
-    dout(10) << __func__ << " readop not complete: " << rop << dendl;
-  } else {
-    dout(10) << __func__ << " readop complete: " << rop << dendl;
+    if (rop.do_redundant_reads) {
+      for (map<hobject_t, read_result_t>::const_iterator iter =
```

Comment:
@athanatos I don't understand why rop.complete is a map in ReadOp.  Because it is, this code that is looking to see if there are enough copies has a loop which is just confusing since I think there is only the need for 1 read_result_t.  Can multiple object read operations all be coming into this same routine?  But the final completion only happens, when all reads have been replied to or with this change enough chunks have been read.